### PR TITLE
Cascade delete on oauth2 table when deleting a user

### PR DESCRIPTION
**Pull request by j0k3r**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | no
| CHANGELOG.md  | no
| License       | MIT

Fix https://github.com/wallabag/wallabag/issues/3883

### CODE REVIEW

**Comment by tcitworld** [comment]
Changes to tests will conflict those in #3955

File: app/DoctrineMigrations/Version20190510141130.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+                $this->addSql('ALTER TABLE ' . $this->getTable('oauth2_auth_codes') . ' DROP CONSTRAINT FK_EE52E3FAA76ED395');
+                $this->addSql('ALTER TABLE ' . $this->getTable('oauth2_auth_codes') . ' ADD CONSTRAINT FK_EE52E3FAA76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
+                break;
+        }
+    }
+
+    public function down(Schema $schema): void
+    {
+        throw new SkipMigrationException('Too complex ...');
```
Comment (tcitworld) [inline app/DoctrineMigrations/Version20190510141130.php]: Well, they could be added for MySQL & PGSQL.

File: app/DoctrineMigrations/Version20190510141130.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+{
+    public function up(Schema $schema): void
+    {
+        switch ($this->connection->getDatabasePlatform()->getName()) {
+            case 'sqlite':
+                $this->addSql('DROP INDEX IDX_368A4209A76ED395');
+                $this->addSql('DROP INDEX IDX_368A420919EB6921');
+                $this->addSql('DROP INDEX UNIQ_368A42095F37A13B');
+                $this->addSql('DROP TABLE ' . $this->getTable('oauth2_access_tokens', true) . '');
+                $this->addSql('CREATE TABLE ' . $this->getTable('oauth2_access_tokens', true) . ' (id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, client_id INTEGER NOT NULL, user_id INTEGER DEFAULT NULL, expires_at INTEGER DEFAULT NULL, token VARCHAR(191) NOT NULL, scope VARCHAR(191) NOT NULL, CONSTRAINT FK_368A420919EB6921 FOREIGN KEY (client_id) REFERENCES ' . $this->getTable('oauth2_clients', true) . ' (id) NOT DEFERRABLE INITIALLY IMMEDIATE, CONSTRAINT FK_368A4209A76ED395 FOREIGN KEY (user_id) REFERENCES "wallabag_user" (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE)');
```
Comment (tcitworld) [inline app/DoctrineMigrations/Version20190510141130.php]: Missing the `__temp__` table creation here.

File: app/DoctrineMigrations/Version20190510141130.php (hunk 1 above)
Comment (j0k3r) [reply app/DoctrineMigrations/Version20190510141130.php]: Yeah but as it has only cascade, I don't think it's necessary

**Pull request merged**